### DISCUSSION TITLE
데이터 구조 논의 쓰레드

### DISCUSSION THREAD

**Discussion by ValseLee** [APPSCHOOL1-REPO/finalproject-gitspace#241] [All About Structure]
- 데이터의 효율적 사용을 위한 리팩토링 논의 쓰레드

**Comment by ValseLee** [comment]
## 각 뷰마다 필요한 데이터
### 노크
- UserID, GitHubUserID, GitHubUserProfileImage(URL), KnockID

**Comment by wontaeyoung** [comment]
### Chat
- UserID, GithubUserID, ChatID, GithubUserProfileImageURL(avatar_url)

### Memo
- 프로필 캐시 이미지를 사용하기 위해서 해당 유저의 githubUserName이 필요한데, 현재 명확하게 저장되어 있는 곳이 Chat 모델의 연산 프로퍼티밖에 없어서 여러 뷰에서 재사용 가능한 형태로 모델링이 되어야합니다.

**Comment by ValseLee** [comment]
**[UserInfo 관련 통합 및 추상화 방향]**

1. Stata를 리턴받는 메소드는 GitHubUserInfo, GSUserInfo 단 2개만 리턴받도록 정의
2. 정의할 때 파라미터는 ID를 필수로 받고, 엔드포인트 파라미터는 옵셔널(기본값은 nil) 로 설정해서 바인딩 여부에 따라 다른 로직이 작동하도록 구성.
3. 가능하다면 프로토콜로 메소드 단순화
